addons/idporten.md

```diff
@@ -0,0 +1,339 @@
+---
+description: >
+  Provisioning and configuration of accompanying ID-porten clients for authentication and authorization in public facing 
+  web applications.
+---
+
+# ID-porten Clients
+
+An accompanying ID-porten Client can be automatically provisioned to your NAIS application. 
+
+The ID-porten Client will be configured with sane defaults to enable usage in both authentication and/or authorization 
+for public facing web applications.
+
+[Digdirator] generates a Kubernetes Secret containing the values needed for your application to integrate ID-porten, 
+e.g. credentials and URLs. The secret will automatically be mounted to the pods of your application during deploy.
+
+Every deploy will trigger rotation of credentials, invalidating any jwk´s that are not in use. 
+_In use_ in this context refers to all jwk´s that are currently mounted to an existing pod - 
+regardless of their status (Running, CrashLoopBackOff, etc.). In other words, jwk rotation should happen
+with zero downtime.
+
+## Spec
+
+See the [NAIS manifest](../nais-application/reference.md#spec-idporten).
+
+## Usage
+
+### ID-porten Client for Authentication & Authorization
+
+Generally, the ID-porten client allows your application to leverage ID-porten for authentication and authorization.
+
+The most common cases include:
+
+- User (citizens) sign-in with SSO, using [OpenID Connect with Authorization Code]
+- Request chains involving an end-user whose identity and permissions should be propagated through each service/web API, 
+using the [OAuth 2.0 Token Exchange].
+
+See the [TokenX Documentation] for NAV-specific usage.
+
+For more information about integration with provider see [ID-porten Integration guide].
+
+### Getting Started
+
+#### Minimal Example
+
+The very minimal example configuration required in [`nais.yaml`](../nais-application/reference.md#spec-idporten)
+to enable auto-provisioning of an ID-porten client for your application.
+
+```yaml
+apiVersion: "nais.io/v1alpha1"
+kind: "Application"
+metadata:
+  name: nais-testapp
+  namespace: aura
+  labels:
+    team: aura
+spec:
+  image: navikt/nais-testapp:66.0.0
+  idporten:
+    enabled: true
+  # required for on-premises only
+  webproxy: true
+  # required for GCP only
+  accessPolicy:
+    outbound:
+      external:
+        - host: oidc-ver2.difi.no
+```
+
+This will register an ID-porten client using the following naming scheme: 
+```
+<cluster>:<metadata.namespace>:<metadata.name>
+```
+
+For the example above, the result would be:
+```
+dev-gcp:aura:nais-testapp
+```
+
+#### Full Example
+
+```yaml
+apiVersion: "nais.io/v1alpha1"
+kind: "Application"
+metadata:
+  name: nais-testapp
+  namespace: aura
+  labels:
+    team: aura
+spec:
+  image: navikt/nais-testapp:66.0.0
+  idporten:
+    enabled: true
+    clientName: NAV - DEV
```

Comment:
Slenge på en kommentar om at clientName må være unikt for hele IDPorten? (for det må det vel være?)

Reply:
Hvis denne må være unik, så tror jeg det kunne være fornuftig å legge inn en verdi som ikke er gyldig yaml, så vi ikke ender opp med klipp/lim uhell.

Reply:
Client_name er ikke unikt

Reply:
Men bør beskrive plassen som sluttbruker prøver å logge seg på. Som NAV eller liknande, denne verdien har en default vill jeg huske hvis man ikke setter den nåe i stil med Arbeids og velferds direktoratet

Reply:
ARBEIDS- OG VELFERDSETATEN 😎